### PR TITLE
Fix Issue 23585 - Add bp relative symbols (frame-relative, see ntreh.…

### PR DESCRIPTION
**Pull request by maxhaton**
…d for apparently the only use) to local symbols during inlining.

### CODE REVIEW

**Comment by maxhaton** [comment]
If it breaks something it'll have to be in the `break`

File: compiler/test/compilable/extra-files/inliner23585b.d [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// https://issues.dlang.org/show_bug.cgi?id=23585
+
+void f()
+{
+    import std.file : exists;
```
Comment (thewilsonator) [inline compiler/test/compilable/extra-files/inliner23585b.d:5]: No std lib in the test suite

**Review by thewilsonator — COMMENTED** [review]
Are you able to verify without using phobos?

**Comment by maxhaton** [comment]
I had a go and failed

**Comment by ibuclaw** [comment]
> I had a go and failed

Try again. ;-)

File: compiler/test/compilable/inliner23585a.d [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// COMPILED_IMPORTS: extra-files/inliner23585b.d
```
Comment (ibuclaw) [inline compiler/test/compilable/inliner23585a.d:1]: This is the wrong directive to use. What you want is `EXTRA_SOURCES`

**Review by ibuclaw — CHANGES_REQUESTED** [review]

**Comment by ibuclaw** [comment]
Didn't #14730 fix the bug?

**Comment by maxhaton** [comment]
I was still able to reproduce the bug locally, I think Razvan put the hack
in the wrong stack. Same reduction possivly though.

Slightly alarmijg that copying a tree is this difficult, deprecate -O.

**Comment by maxhaton** [comment]
Previous one did get I think, closing.

**Pull request closed without merging**